src/magic_folder/test/test_web.py

```diff
@@ -266,6 +281,8 @@ def treq_for_folders(reactor, basedir, auth_token, folders, start_folder_service
     :param bool start_folder_services: If ``True``, start the Magic Folder
         service objects.  Otherwise, don't.
 
+    :param TahoeClient tahoe_client: if provided, used as the tahoe-client
+
```

Comment:
Yeah we probably do need to return more than the treq stand-in .. the thing we need to call `stopService()` on is the `global_service` instance.

Conceptually it might be possible to set up the correct tahoe data for the given "folders" mapping .. but I feel like it's probably more-explicit/easier to read if that setup happens in the test itself?

(`test_add_participant` is a test that actually sets up some Tahoe data, for comparison .. so I guess the alternative is to provide a `tahoe_client_for_participants()` or similar that sets up the Collective DMD properly and move those lines into that function instead...?)

Reply:
> Yeah we probably do need to return more than the treq stand-in .. the thing we need to call stopService() on is the global_service instance.

I think this isn't true, given that we don't start it, but only some children.

> Conceptually it might be possible to set up the correct tahoe data for the given "folders" mapping .. but I feel like it's probably more-explicit/easier to read if that setup happens in the test itself?
>
> (test_add_participant is a test that actually sets up some Tahoe data, for comparison .. so I guess the alternative is to provide a tahoe_client_for_participants() or similar that sets up the Collective DMD properly and move those lines into that function instead...?)

I don't have strong feelings on this either way. We can keep the code in the test, but updating `tahoe_root._uri.data` after the fact (or, we could thread passing that value through, and expose it as a parameter).

Reply:
> I think this isn't true, given that we don't start it, but only some children.

yes, that's more-accurate ;) but we do get the sub-services from it..